tests/unit_tests/c_wrappers/res/enkf/config/test_gen_kw_config.py

```diff
@@ -199,3 +200,47 @@ def test_gen_kw_is_log_or_not(
                 encoding="utf-8"
             ),
         )
+
+
+@pytest.mark.parametrize(
+    "distribution, mean, std, error",
+    [
+        ("LOGNORMAL", "0", "1", None),
+        ("LOGNORMAL", "-1", "1", ["MEAN"]),
+        ("LOGNORMAL", "0", "-1", ["STD"]),
+        ("LOGNORMAL", "-1", "-1", ["MEAN", "STD"]),
+        ("NORMAL", "0", "1", None),
+        ("NORMAL", "-1", "1", None),
+        ("NORMAL", "0", "-1", ["STD"]),
+        ("TRUNCATED_NORMAL", "-1", "1", None),
+        ("TRUNCATED_NORMAL", "0", "1", None),
+        ("TRUNCATED_NORMAL", "0", "-1", ["STD"]),
+    ],
+)
+def test_gen_kw_distribution_errors(tmpdir, distribution, mean, std, error):
+    with tmpdir.as_cwd():
+        config = dedent(
+            """
+        JOBNAME my_name%d
+        NUM_REALIZATIONS 1
+        GEN_KW KW_NAME template.txt kw.txt prior.txt
+        """
+        )
+        with open("config.ert", "w", encoding="utf-8") as fh:
+            fh.writelines(config)
+        with open("template.txt", "w", encoding="utf-8") as fh:
+            fh.writelines("MY_KEYWORD <MY_KEYWORD>")
+        with open("prior.txt", "w", encoding="utf-8") as fh:
+            if distribution == "TRUNCATED_NORMAL":
+                fh.writelines(f"MY_KEYWORD {distribution} {mean} {std} -1 1")
```

Comment:
I guess testing of the hard limits for `TRUNCATED_NORMAL` is out of scope for this?
Maybe it does not make sense to test those?

Reply:
Not really sure what are the hard limits of `TRUNCATED_NORMAL` to actually test them, but that is not the purpose of this test.